### PR TITLE
mds/OpenFileTable: add _read_omap_values() to simplify the code

### PR DESCRIPTION
**Pull request by lxbsz**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/mds/OpenFileTable.cc [hunk 1]
```diff
@@ -725,6 +725,21 @@ void OpenFileTable::_recover_finish(int r)
   waiting_for_load.clear();
 }
 
+void OpenFileTable::_read_omap_values(const std::string& key, unsigned idx,
```
Comment (batrick) [inline src/mds/OpenFileTable.cc:728]:
```suggestion
void OpenFileTable::_read_omap_values(std::string_view key, unsigned idx,
```

**Review by batrick — CHANGES_REQUESTED** [review]

File: src/mds/OpenFileTable.cc (hunk 1 above)
Comment (lxbsz) [reply src/mds/OpenFileTable.cc]: Fixed it.

It intended to be comatiable with the `op.omap_get_vals(const std::string &key, ...)`.

File: src/mds/OpenFileTable.cc [hunk 2]
```diff
@@ -841,17 +856,11 @@ void OpenFileTable::_load_finish(int op_r, int header_r, int values_r,
...
-    C_IO_OFT_Load *c = new C_IO_OFT_Load(this, idx, !more);
-    ObjectOperation op;
-    if (!more)
-      op.omap_get_header(&c->header_bl, &c->header_r);
-    op.omap_get_vals(last_key, "", uint64_t(-1),
-		     &c->values, &c->more, &c->values_r);
-    mds->objecter->read(oid, oloc, op, CEPH_NOSNAP, nullptr, 0,
-			new C_OnFinisher(c, mds->finisher));
+    dout(10) << __func__ << ": continue to load from '"
+             << get_object_name(idx) << ":" << last_key << "'"
```
Comment (joscollin) [inline src/mds/OpenFileTable.cc:860]: Now `get_object_name(idx)` is called 2 times in this code path?

File: src/mds/OpenFileTable.cc (hunk 1 above)
Comment (batrick) [reply src/mds/OpenFileTable.cc]: I see. The change I suggested will end up creating unnecessary string temporaries. Let's undo this.

Comment (lxbsz) [reply src/mds/OpenFileTable.cc]: Done.

File: src/mds/OpenFileTable.cc (hunk 2 above)
Comment (lxbsz) [reply src/mds/OpenFileTable.cc]: Yeah, moved it to `_read_omap_values()` to fix this.

**Review by joscollin — APPROVED** [review]

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201214.211711

**Pull request merged**